Source/missiles.cpp

```diff
@@ -915,6 +874,105 @@ Direction16 GetDirection16(Point p1, Point p2)
 	return ret;
 }
 
+void MissileHitMonsterConsequences(int mid, int pnum, int dam, missile_id mName)
+{
+	auto &monster = Monsters[mid];
+	const auto &player = Players[pnum];
+
+	bool hasKnockback = false;
+	if (pnum >= 0) {
+		const auto &player = Players[pnum];
+		if (HasAnyOf(player._pIFlags, ItemSpecialEffect::Knockback))
+			hasKnockback = true;
+	}
+
+	if (monster._mhitpoints >> 6 <= 0) {
+		M_StartKill(mid, pnum);
+	} else if (monster.IsResistant(mName)) {
+		PlayEffect(monster, 1);
+	} else {
+		if (monster._mmode != MonsterMode::Petrified
+		    && MissilesData[mName].mType == 0
+		    && hasKnockback)
+			M_GetKnockback(mid);
+		if (monster.MType->mtype != MT_GOLEM)
+			M_StartHit(mid, pnum, dam);
+	}
+}
+
+int PlayerMissileCalculateCTHAgainstMonster(int pnum, missile_id mName, int dist, Monster &monster)
+{
+	int hper = 0;
+	if (pnum != -1) {
+		const auto &player = Players[pnum];
+		if (MissilesData[mName].mType == 0) {
+			hper = player.GetRangedPiercingToHit();
+			hper -= player.CalculateArmorPierce(monster.mArmorClass, false);
+			hper -= (dist * dist) / 2;
+		} else {
+			hper = player.GetMagicToHit() - (monster.mLevel * 2) - dist;
+		}
+	} else {
+		hper = GenerateRnd(75) - monster.mLevel * 2;
+	}
+	return hper;
+}
+
+int PlayerMissileCalculateDamageAgainstMonster(int pnum, int minDam, int maxDam, missile_id mName)
+{
+	int dam = minDam + GenerateRnd(maxDam - minDam + 1);
+
+	const auto &player = Players[pnum];
+
+	if (MissilesData[mName].mType == 0 && MissilesData[mName].mResist == MISR_NONE) {
+		dam = player._pIBonusDamMod + dam * player._pIBonusDam / 100 + dam;
+		if (player._pClass == HeroClass::Rogue)
+			dam += player._pDamageMod;
+		else
+			dam += player._pDamageMod / 2;
+	}
+	return dam;
+}
+
+void PlayerMissileHitMonster(int pnum, int dam, int m, missile_id mName)
+{
+	auto &monster = Monsters[m];
+	if (pnum == MyPlayerId)
+		monster._mhitpoints -= dam;
+
+	const auto &player = Players[pnum];
+	if ((gbIsHellfire && HasAnyOf(player._pIFlags, ItemSpecialEffect::NoHealOnMonsters)) || (!gbIsHellfire && HasAnyOf(player._pIFlags, ItemSpecialEffect::FireArrows)))
+		monster._mFlags |= MFLAG_NOHEAL;
+
+	MissileHitMonsterConsequences(m, pnum, dam, mName);
+
+	if (monster._msquelch == 0) {
+		monster._msquelch = UINT8_MAX;
+		monster.position.last = player.position.tile;
+	}
+}
+
+int TrapMissileCalculateCTHAgainstMonster(int dist, Monster &monster)
+{
+	return 90 - (BYTE)monster.mArmorClass - dist;

```

Comment:
The old WINAPI C-style cast can be removed as `mArmorClass` is defined as `uint8_t`.
```suggestion
	return 90 - monster.mArmorClass - dist;
```

Reply:
this is probably resolved now